test/makeTravisReport.m

```diff
@@ -1,45 +1,260 @@
+    reportUnreliableTests(arg, S);
+    reportReliableTests(arg, S);
+    displayTestSummary(arg.stream, S);
+
+    %% set output arguments if needed
     if nargout >= 1
-        nErrors = countNumberOfErrors(reliableTests);
+        nErrors = countNumberOfErrors(S.reliable);
     end
 end
-% ==============================================================================
-function displaySummaryTable(stream, status)
-    % display a summary table of all tests
-    for iTest = 1:numel(status)
-        fprintf(stream, '%s\n', formatSummaryRow(status{iTest}));
+% == INPUT VALIDATOR FUNCTIONS =================================================
+function bool = isstream(val)
+    % returns true if it is a valid (file) stream or stdout/stderr stream
+    bool = isnumeric(val);
+end
+function bool = islength(val)
+    % validates the report length
+    bool = ismember(lower(val), {'default','short','long'});
+end
+% == GITHUB-FLAVORED MARKDOWN FUNCTIONS ========================================
+function str = gfmTable(data, header, alignment)
+    % Construct a Github-flavored Markdown table
+    %
+    % Arguments:
+    %   - data: nRows x nCols cell array that represents the data
+    %   - header: cell array with the (nCol) column headers
+    %   - alignment: alignment specification per column
+    %       * 'l': left-aligned (default)
+    %       * 'c': centered
+    %       * 'r': right-aligned
+    %     When not enough entries are specified, the specification is repeated
+    %     cyclically.
+    %
+    % Output: table as a string
+    %
+    % See https://help.github.com/articles/github-flavored-markdown/#tables
+
+    % input argument validation and normalization
+    nCols = size(data, 2);
+    if ~exist('alignment','var') || isempty(alignment)
+        alignment = 'l';
+    end
+    if numel(alignment) < nCols
+        % repeat the alignment specifications to along the columns
+        alignment = repmat(alignment, 1, nCols);
+        alignment = alignment(1:nCols);
     end
 
-    nErrors = countNumberOfErrors(status);
-    if nErrors > 0
-        fprintf(stream,'\n%3d of %3d tests failed. :-( \n', nErrors, numel(status));
+    % calculate the required column width
+    cellWidth   = cellfun(@length, [header(:)' ;data]);
+    columnWidth = max(max(cellWidth, [], 1),3); % use at least 3 places
+
+    % prepare the table format
+    COLSEP = '|'; ROWSEP = sprintf('\n');
+    rowformat = [COLSEP sprintf([' %%%ds ' COLSEP], columnWidth) ROWSEP];
+    alignmentRow = formatAlignment(alignment, columnWidth);
+
+    % actually print the table
+    fullTable = [header; alignmentRow; data];
+    strs = cell(size(fullTable,1), 1);
+    for iRow = 1:numel(strs)
+        thisRow = fullTable(iRow,:);
+        %TODO: maybe preprocess thisRow with strjust first
+        strs{iRow} = sprintf(rowformat, thisRow{:});
+    end
+    str = [strs{:}];
+    function alignRow = formatAlignment(alignment, columnWidth)
```

Comment:
should some kind of delimeter line be added between inline functions (also above and below)?